grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationAtom.java

```diff
@@ -225,7 +225,7 @@ public boolean isAllowedToFormRuleHead(){
         }
 
         //check roles are ok
-        Collection<Role> possibleRoles = type != null? type.asRelationType().relates() : Collections.EMPTY_SET;
+        Set<Role> possibleRoles = type != null? type.asRelationType().relates().collect(toSet()) : Collections.EMPTY_SET;
```

Comment:
I think I tried to but I was not able to do so. Will try again. Maybe It was something silly I was doing during the original attempt to change everything.